go/libkb/sig_chain.go

```diff
@@ -343,6 +348,12 @@ func (sc *SigChain) GetCurrentSubchain(eldest keybase1.KID) (links []*ChainLink,
 	l := len(sc.chainLinks)
 	lastGood := l
 	for i := l - 1; i >= 0; i-- {
+
+		if sc.chainLinks[i].IsStubbed() {
+			lastGood = i
+			continue
+		}
```

Comment:
Hmm we have those three cases to consider. From the code:

```
// GetCurrentSubchain takes the given sigchain and walks backward until it
// finds the start of the current subchain, returning all the links in the
// subchain. A new subchain starts in one of three ways (from the perspective
// of walking from oldest to newest):
// 1) A link has a new eldest key, usually in the form of reporting no
//    eldest_kid of its own and being signed by a KID that's not the previous
//    eldest. Most resets so far take this form.
// 2) A link of type "eldest", regardless of the KIDs involved. We want this to
//    be how everything works in the future.
// 3) One of a set of six hardcoded links that made it in back when we allowed
//    repeating eldest keys without using the "eldest" link type.
```

Is it safe to say that "all sigchain resets after the introduction of v2 chainlinks must be of type eldest" so that if a V2 link exists at all (as opposed to the old-style-v1-only links), we know it's not a reset just based on the type?